diff --git a/BUILD b/BUILD
index bc013c9..d0f4540 100755
--- a/BUILD
+++ b/BUILD
@@ -104,6 +104,9 @@ genrule(
         "//third_party/googleapis:MODULE.bazel",
         "//third_party/remoteapis:MODULE.bazel",
         "//third_party:BUILD",
+        "//third_party:platforms_0.0.10.patch",
+        "//third_party:rules_go_0.39.1.patch",
+        "//third_party:rules_java_7.6.5.patch",
         "//third_party:rules_jvm_external_6.0.patch",
         "//third_party/upb:BUILD",
         "//third_party/upb:01_remove_werror.patch",
diff --git a/MODULE.bazel b/MODULE.bazel
index 883b22d..3ba715f 100755
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -410,3 +410,18 @@ use_repo(bazel_android_deps, "desugar_jdk_libs")
 
 remote_android_extensions = use_extension("//tools/android:android_extensions.bzl", "remote_android_tools_extensions")
 use_repo(remote_android_extensions, "android_gmaven_r8", "android_tools")
+single_version_override(
+    module_name = "platforms",
+    patch_strip = 1,
+    patches = ["//third_party:platforms_0.0.10.patch"],
+)
+single_version_override(
+    module_name = "rules_java",
+    patch_strip = 1,
+    patches = ["//third_party:rules_java_7.6.5.patch"],
+)
+single_version_override(
+    module_name = "rules_go",
+    patch_strip = 1,
+    patches = ["//third_party:rules_go_0.39.1.patch"],
+)
diff --git a/src/conditions/BUILD b/src/conditions/BUILD
index 6fd49f3..89d7428 100755
--- a/src/conditions/BUILD
+++ b/src/conditions/BUILD
@@ -233,3 +233,11 @@ config_setting(
     },
     visibility = ["//visibility:public"],
 )
+config_setting(
+    name = "linux_loongarch64",
+    constraint_values = [
+        "@platforms//os:linux",
+        "@platforms//cpu:loongarch64",
+    ],
+    visibility = ["//visibility:public"],
+)
diff --git a/src/conditions/BUILD.tools b/src/conditions/BUILD.tools
index aafdb4f..c2f9b18 100755
--- a/src/conditions/BUILD.tools
+++ b/src/conditions/BUILD.tools
@@ -157,3 +157,11 @@ config_setting(
     },
     visibility = ["//visibility:public"],
 )
+config_setting(
+    name = "linux_loongarch64",
+    constraint_values = [
+        "@platforms//os:linux",
+        "@platforms//cpu:loongarch64",
+    ],
+    visibility = ["//visibility:public"],
+)
diff --git a/src/main/cpp/archive_utils.h b/src/main/cpp/archive_utils.h
index 18856ab..3c8529f 100755
--- a/src/main/cpp/archive_utils.h
+++ b/src/main/cpp/archive_utils.h
@@ -21,6 +21,7 @@
 #include "src/main/cpp/startup_options.h"
 #include "src/main/cpp/util/logging.h"
 
+#include <cstdint>
 namespace blaze {
 
 // Determines the contents of the archive, storing the names of the contained
diff --git a/src/main/cpp/blaze.h b/src/main/cpp/blaze.h
index 2350fa1..1a6bc8a 100755
--- a/src/main/cpp/blaze.h
+++ b/src/main/cpp/blaze.h
@@ -20,6 +20,7 @@
 #include "src/main/cpp/workspace_layout.h"
 #include "src/main/cpp/util/logging.h"
 
+#include <cstdint>
 namespace blaze {
 
 // Prints client version information to standard output, e.g. when invoking the
diff --git a/src/main/java/com/google/devtools/build/lib/analysis/config/AutoCpuConverter.java b/src/main/java/com/google/devtools/build/lib/analysis/config/AutoCpuConverter.java
index f768df9..e459edf 100755
--- a/src/main/java/com/google/devtools/build/lib/analysis/config/AutoCpuConverter.java
+++ b/src/main/java/com/google/devtools/build/lib/analysis/config/AutoCpuConverter.java
@@ -72,6 +72,8 @@ public class AutoCpuConverter extends Converter.Contextless<String> {
               return "mips64";
             case RISCV64:
               return "riscv64";
+            case LOONGARCH64:
+              return "loongarch64";
             default:
               return "unknown";
           }
diff --git a/src/main/java/com/google/devtools/build/lib/util/CPU.java b/src/main/java/com/google/devtools/build/lib/util/CPU.java
index 88ae4a3..87f7dde 100755
--- a/src/main/java/com/google/devtools/build/lib/util/CPU.java
+++ b/src/main/java/com/google/devtools/build/lib/util/CPU.java
@@ -29,6 +29,7 @@ public enum CPU {
   S390X("s390x", ImmutableSet.of("s390x", "s390")),
   MIPS64("mips64", ImmutableSet.of("mips64el", "mips64")),
   RISCV64("riscv64", ImmutableSet.of("riscv64")),
+  LOONGARCH64("loongarch64", ImmutableSet.of("loong64", "loongarch64")),
   UNKNOWN("unknown", ImmutableSet.<String>of());
 
   private final String canonicalName;
diff --git a/third_party/platforms_0.0.10.patch b/third_party/platforms_0.0.10.patch
new file mode 100644
index 0000000..cb6f5ee
--- /dev/null
+++ b/third_party/platforms_0.0.10.patch
@@ -0,0 +1,29 @@
+diff --git a/cpu/BUILD b/cpu/BUILD
+index 6497452..5eb70fc 100644
+--- a/cpu/BUILD
++++ b/cpu/BUILD
+@@ -188,3 +188,11 @@ constraint_value(
+     name = "riscv64",
+     constraint_setting = ":cpu",
+ )
++alias(
++    name = "loong64",
++    actual = ":loongarch64",
++)
++constraint_value(
++    name = "loongarch64",
++    constraint_setting = ":cpu",
++)
+diff --git a/host/extension.bzl b/host/extension.bzl
+index 8756c59..ae4004d 100644
+--- a/host/extension.bzl
++++ b/host/extension.bzl
+@@ -15,6 +15,8 @@ def _translate_cpu(arch):
+         return "mips64"
+     if arch in ["riscv64"]:
+         return "riscv64"
++    if arch in ["loongarch64"]:
++        return "loongarch64"
+     return None
+ 
+ def _translate_os(os):
diff --git a/third_party/rules_go_0.39.1.patch b/third_party/rules_go_0.39.1.patch
new file mode 100644
index 0000000..a95e97e
--- /dev/null
+++ b/third_party/rules_go_0.39.1.patch
@@ -0,0 +1,41 @@
+diff --git a/go/private/platforms.bzl b/go/private/platforms.bzl
+index 4da4306..2374a42 100644
+--- a/go/private/platforms.bzl
++++ b/go/private/platforms.bzl
+@@ -26,6 +26,7 @@ BAZEL_GOOS_CONSTRAINTS = {
+ }
+ 
+ BAZEL_GOARCH_CONSTRAINTS = {
++    "loong64": "@platforms//cpu:loongarch64",
+     "386": "@platforms//cpu:x86_32",
+     "amd64": "@platforms//cpu:x86_64",
+     "arm": "@platforms//cpu:arm",
+@@ -65,6 +66,7 @@ GOOS_GOARCH = (
+     ("linux", "ppc64"),
+     ("linux", "ppc64le"),
+     ("linux", "riscv64"),
++    ("linux", "loong64"),
+     ("linux", "s390x"),
+     ("nacl", "386"),
+     ("nacl", "amd64p32"),
+@@ -124,6 +126,7 @@ CGO_GOOS_GOARCH = {
+     ("linux", "mipsle"): None,
+     ("linux", "ppc64le"): None,
+     ("linux", "riscv64"): None,
++    ("linux", "loong64"): None,
+     ("linux", "s390x"): None,
+     ("linux", "sparc64"): None,
+     ("netbsd", "386"): None,
+diff --git a/go/private/sdk.bzl b/go/private/sdk.bzl
+index 5a3ef42..30c88fd 100644
+--- a/go/private/sdk.bzl
++++ b/go/private/sdk.bzl
+@@ -478,6 +478,8 @@ def _detect_host_platform(ctx):
+         goarch = "arm64"
+     elif goarch == "x86_64":
+         goarch = "amd64"
++    elif goarch == "loongarch64":
++        goarch = "loong64"
+ 
+     return goos, goarch
+ 
diff --git a/third_party/rules_java_7.6.5.patch b/third_party/rules_java_7.6.5.patch
new file mode 100644
index 0000000..4eb73c4
--- /dev/null
+++ b/third_party/rules_java_7.6.5.patch
@@ -0,0 +1,20 @@
+diff --git a/toolchains/BUILD b/toolchains/BUILD
+index 81126c2..691a005 100644
+--- a/toolchains/BUILD
++++ b/toolchains/BUILD
+@@ -126,6 +126,7 @@ cc_library(
+         "@bazel_tools//src/conditions:linux_mips64": [":jni_md_header-linux"],
+         "@bazel_tools//src/conditions:linux_ppc64le": [":jni_md_header-linux"],
+         "@bazel_tools//src/conditions:linux_riscv64": [":jni_md_header-linux"],
++        "@bazel_tools//src/conditions:linux_loongarch64": [":jni_md_header-linux"],
+         "@bazel_tools//src/conditions:linux_s390x": [":jni_md_header-linux"],
+         "@bazel_tools//src/conditions:linux_x86_64": [":jni_md_header-linux"],
+         "@bazel_tools//src/conditions:openbsd": [":jni_md_header-openbsd"],
+@@ -136,6 +137,7 @@ cc_library(
+         "@bazel_tools//src/conditions:darwin": ["include/darwin"],
+         "@bazel_tools//src/conditions:freebsd": ["include/freebsd"],
+         "@bazel_tools//src/conditions:linux_aarch64": ["include/linux"],
++        "@bazel_tools//src/conditions:linux_loongarch64": ["include/linux"],
+         "@bazel_tools//src/conditions:linux_mips64": [":include/linux"],
+         "@bazel_tools//src/conditions:linux_ppc64le": ["include/linux"],
+         "@bazel_tools//src/conditions:linux_riscv64": [":include/linux"],
diff --git a/tools/cpp/lib_cc_configure.bzl b/tools/cpp/lib_cc_configure.bzl
index 67ee9ca..5e6f412 100755
--- a/tools/cpp/lib_cc_configure.bzl
+++ b/tools/cpp/lib_cc_configure.bzl
@@ -208,6 +208,8 @@ def get_cpu_value(repository_ctx):
         return "mips64"
     if arch in ["riscv64"]:
         return "riscv64"
+    if arch in ["loong64", "loongarch64"]:
+        return "loongarch64"
     if arch in ["arm", "armv7l"]:
         return "arm"
     if arch in ["aarch64"]:
